packages/ui/src/elements/settings-overlay.tsx

```diff
@@ -167,41 +167,71 @@ const ChangeButton = styled.div((props: EditorThemeProps) => {
   }
 })
 
-export const InlineOverlay: React.FunctionComponent<{
+export const InlineSettings: React.FunctionComponent<{
   onEdit: React.MouseEventHandler
   onDelete: React.MouseEventHandler
+}> = props => {
+  return (
+    <HoveringOverlay position={'below'}>
+      <InlinePreview>{props.children}</InlinePreview>
+      <ChangeButton onClick={props.onEdit}>
+        <Icon icon={faPencilAlt} />
+      </ChangeButton>
+      <ChangeButton onClick={props.onDelete}>
+        <Icon icon={faTrashAlt} />
+      </ChangeButton>
+    </HoveringOverlay>
+  )
+}
+
+export type HoverPosition = 'above' | 'below'
+
+export const HoveringOverlay: React.FunctionComponent<{
+  position: HoverPosition
 }> = props => {
   const overlay = React.createRef<HTMLDivElement>()
   React.useEffect(() => {
     const menu = overlay.current
     if (!menu) return
-
     const native = window.getSelection()
     const range = native.getRangeAt(0)
     const rect = range.getBoundingClientRect()
     if (rect.height === 0) return
     // menu is set to display:none, shouldn't ever happen
     if (!menu.offsetParent) return
     const parentRect = menu.offsetParent.getBoundingClientRect()
+    // only show menu if selection is inside of parent
+    if (
+      parentRect.top - 5 > rect.top ||
+      parentRect.top + parentRect.height + 5 < rect.top + rect.height ||
+      parentRect.left - 5 > rect.left ||
+      parentRect.left + parentRect.width + 5 < rect.left + rect.width
+    ) {
+      menu.style.top = '-10000px'
+      menu.style.left = '-10000px'
```

Comment:
you could reset the style using `menu.style = undefined`, so only the styled component is applied.

Reply:
thanks, resolved